### PR TITLE
Add workflow dispatch and allow workflow names.

### PR DESCRIPTION
**Pull request by fodinabor**
Adds the workflow dispatch proposed in #939.
This is different from the repository dispatch, as here it is clear what arguments to use, so that this is much more convenient to use.

Also update docs link.

### CODE REVIEW

**Comment by bummzack** [comment]
I'm looking for this functionality as well. Any chance to get this merged? 
Or are there currently other ways to trigger a [workflow-dispatch](https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event) event?

**Comment by bummzack** [comment]
I've tested using this PR, but there's an issue when dispatching without the optional `$inputs` param.

> For 'properties/inputs', [] is not an object.

I think inputs should be omitted if there's no input to be specified… this works:

```php
public function dispatches(string $username, string $repository, string $workflow, string $ref, array $inputs = null)
{
    $parameters = array_filter(['ref' => $ref, 'inputs' => $inputs]);

    return $this->post('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/actions/workflows/'.rawurlencode($workflow).'/dispatches', $parameters);
}
```

File: lib/Github/Api/Repository/Actions/Workflows.php
Comment (acrobat) [inline lib/Github/Api/Repository/Actions/Workflows.php:49]: This is actually a BC break, as users who pass an integer will get a TypeError now. Can you remove the string typehint so we allow both int and string values (but add a `@param` doc for this parameter to defined the allowed types). And check in the method itself if it's a string, then rawurlencode the value otherwise just pass it to the url.

File: lib/Github/Api/Repository/Actions/Workflows.php
Comment (acrobat) [inline lib/Github/Api/Repository/Actions/Workflows.php:65]: Maybe use the same approach here for the `$workflow` parameter, to allow both int and string. When bumping php versions we can typehint this with `string|int`.

**Review by acrobat — CHANGES_REQUESTED** [review]
Thanks for the PR! 1 general remark/question, can you remove the `@link` url changes and move them to a seperate PR as they are not related to this specific feature. This will allow the git history to give a better insight in why a certain change was done. Thanks!

Apart from 2 comments on the dispatch methods, this pr looks good! 👌

**Comment by fodinabor** [comment]
should be non-BC breaking now (even though Roave still doesn't like it), will wait for the URL update until this is merged

**Comment by acrobat** [comment]
Thanks @fodinabor! And congrats on your first contribution! 🎉

**Pull request merged**
